crm_salesperson_planner/models/calendar_event.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Sygel - Manuel Regidor
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class CalendarEvent(models.Model):
+    _inherit = "calendar.event"
+
+    def write(self, values):
+        ret_val = super(CalendarEvent, self).write(values)
+        if ret_val and values.get("start"):
+            for sel in self.filtered(
+                lambda a: a.res_model == "crm.salesperson.planner.visit"
+            ):
+                salesperson_visit_id = self.env["crm.salesperson.planner.visit"].search(
```

Comment:
Do browse over `res_id` instead of the search. Other possibility is to add a one2many here and directly use `mapped`.

Another thing: as the start value is the same for all, please do only one write.

Reply:
Done.